src/Common/ThreadPool.cpp

```diff
@@ -276,7 +277,11 @@ std::unique_ptr<GlobalThreadPool> GlobalThreadPool::the_instance;
 
 void GlobalThreadPool::initialize(size_t max_threads)
 {
-    assert(!the_instance);
```

Comment:
Why `assert()` is not enough in this particular case?

Reply:
I want to see whether it reproduces in integration tests, which are not ran in debug.